tests/components/tplink_omada/test_switch.py

```diff
@@ -33,6 +41,77 @@ async def test_poe_switches(
         )
 
 
+async def test_gateway_connect_ipv4_switch(
+    hass: HomeAssistant,
+    mock_omada_site_client: MagicMock,
+    init_integration: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test gateway connected switches."""
+    gateway_mac = "AA-BB-CC-DD-EE-FF"
+
+    entity_id = "switch.test_router_port_4_internet_connected"
+    entity = hass.states.get(entity_id)
+    assert entity == snapshot
+
+    test_gateway = await mock_omada_site_client.get_gateway(gateway_mac)
+    port_status = test_gateway.port_status[3]
+    assert port_status.port_number == 4
+
+    mock_omada_site_client.set_gateway_wan_port_connect_state.reset_mock()
+    mock_omada_site_client.set_gateway_wan_port_connect_state.return_value = (
+        _get_updated_gateway_port_status(
+            mock_omada_site_client, test_gateway, 3, "internetState", 0
+        )
+    )
+    await call_service(hass, "turn_off", entity_id)
+    mock_omada_site_client.set_gateway_wan_port_connect_state.assert_called_once_with(
+        4, False, test_gateway, ipv6=False
+    )
+    entity = hass.states.get(entity_id)
+    assert entity.state == "off"
+
+    mock_omada_site_client.set_gateway_wan_port_connect_state.reset_mock()
+    mock_omada_site_client.set_gateway_wan_port_connect_state.return_value = (
+        _get_updated_gateway_port_status(
+            mock_omada_site_client, test_gateway, 3, "internetState", 1
+        )
+    )
+    await call_service(hass, "turn_on", entity_id)
+    mock_omada_site_client.set_gateway_wan_port_connect_state.assert_called_once_with(
+        4, True, test_gateway, ipv6=False
+    )
+    entity = hass.states.get(entity_id)
+    assert entity.state == "on"
+
+
+async def test_gateway_api_fail_disables_switch_entities(
+    hass: HomeAssistant,
+    mock_omada_site_client: MagicMock,
+    init_integration: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test gateway connected switches."""
+    entity_id = "switch.test_router_port_4_internet_connected"
+    entity = hass.states.get(entity_id)
+    assert entity == snapshot
+    assert entity.state == "on"
+
+    controller: OmadaSiteController = hass.data[init_integration.domain][
```

Comment:
Never access hass.data in tests. That's an integration detail.

Reply:
Fixed. Moving time forward made it unnecessary anyway.